atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/config/ConnectionConfig.java

```diff
@@ -58,7 +58,10 @@ public MaskedValue getDbPassword() {
 
     public abstract String getDriverClass();
 
-    public abstract String getTestQuery();
+    @Value.Default
+    public String getTestQuery() {
```

Comment:
This change brings Atlas into alignment with Hikari's recommendation to use the JDBC drivers isValid() method. However, we've also seen that (on Oracle at least) isValid() can be a pretty expensive query depending on the connection state - it spins up a transient thread.

Maybe the better change would be to just update OracleConnectionConfig to return "select 1 from dual"

Reply:
I'd slightly prefer to get some data before deviating from the recommended default. Although the Oracle JDBC driver will create a new thread because the validation timeout is greater than zero, it's not clear how much of a performance impact this is.

When testing this internally, we only disabled `testConnectionBeforeHandout` and began seeing adequate performance. We may want to do some more internal testing where we explicitly set `testQuery` to see if that result in further performance improvements.